modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreator.java

```diff
@@ -160,10 +160,18 @@ public Subject createSubject(SecurityDomainContext ctx, String name, String pass
       if (TRACE)
          SECURITY_LOGGER.aboutToAuthenticate(ctx.getSecurityDomain());
 
+      RealmIdentity identity = null;
       if (securityDomain != null) {
-          // use elytron
+         // elytron security domain
+         try {
+            identity = securityDomain.getIdentity(principal.getName());
+         } catch (RealmUnavailableException e) {
+            throw MESSAGES.authenticationFailed(principal.getName());
+         }
+      }
+      if (identity != null && !identity.getClass().getName().equals("org.jboss.as.security.elytron.SecurityDomainContextRealm$PicketBoxBasedIdentity")) {
```

Comment:
@Skyllarr minor thing. Is it better check this with `identity.getClass().getSimpleName().equals("PicketBoxBasedIdentity")` ?
`PicketBoxBasedIdentity` is a private class in WFLY. Is there any other approach to check Elytron using a legacy jaas picketbox security realm ?

Reply:
@jimma I did not find any better way to check this. Here are the docs https://docs.wildfly.org/27/WildFly_Elytron_Security.html#Properties_File_Based_Authentication_Migration . In this case picketbox exposes its security domain which elytron considers to be a security realm. 

Elytron's security domain can have multiple different security realms that contain users. So I cannot use any check before the identity is actually obtained ( `securityDomain.getIdentity(principal.getName())` ) . So I have to check the type of identity or where it was obtained from. But where it was obtained from is a usual security realm, there is no public API I could use to check what type of realm it is. So I need to check the type of identity.

I noticed the changes from this PR are not yet in the main branch. Is it planned to be ported there as well? Also, just to harden this part of code agains the possible NPEs I submitted a new PR to the main branch https://github.com/jbossws/jbossws-cxf/pull/267 . It could be backported here as well just to harden the code further

Reply:
> I noticed the changes from this PR are not yet in the main branch. Is it planned to be ported there as well? 

I don't think we need to backport this change to main branch. After WFLY25, this picketbox legacy support is already removed from security subsystem. Hence this Identity can't be PicketBoxBasedIdentity any more in WFLY27 and WFLY28.

Reply:
@jimma yes, makes sense, thanks!